### PR TITLE
Accept 'OS_PROJECT_ID' and 'OS_TENANT_ID' credentail env vars.

### PR DESCRIPTION
**Pull request by ExternalReality**

### CODE REVIEW

File: identity/identity.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ type Credentials struct {
 	Secrets       string // The secrets to pass
 	Region        string // Region to send requests to
 	TenantName    string // The project name for this connection
+	TenantId      string `credentials:"optional"` // The project id for this connection
```
Comment (wallyworld) [inline identity/identity.go]: this isn't needed

File: identity/identity_test.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ func (s *CredentialsTestSuite) TestCompleteCredentialsFromEnvDefaultDomain(c *gc
 		"OS_PASSWORD":            "test-pass",
 		"OS_ACCESS_KEY":          "test-access-key",
 		"OS_SECRET_KEY":          "test-secret-key",
-		"OS_PROJECT_NAME":        "tenant-name",
+		"OS_PROJECT_ID":          "tenant-name",
```
Comment (wallyworld) [inline identity/identity_test.go]: this isn't right - we need to test that either OS_PROJECT_NAME or OS_PROJECT_ID works.

File: identity/identity_test.go [hunk 3]
```diff
@@ -134,12 +134,12 @@ func (s *CredentialsTestSuite) TestCompleteCredentialsFromEnvDefaultDomain(c *gc
...
-		"OS_AUTH_URL":     "http://auth",
-		"OS_USERNAME":     "test-user",
-		"OS_ACCESS_KEY":   "test-access-key",
-		"OS_PROJECT_NAME": "tenant-name",
-		"OS_REGION_NAME":  "region",
-		"OS_DOMAIN_NAME":  "domain-name",
+		"OS_AUTH_URL":    "http://auth",
+		"OS_USERNAME":    "test-user",
+		"OS_ACCESS_KEY":  "test-access-key",
+		"OS_TENANT_NAME": "tenant-name",
```
Comment (wallyworld) [inline identity/identity_test.go]: no need to change this

File: identity/identity_test.go [hunk 4]
```diff
@@ -180,7 +180,7 @@ func (s *CredentialsTestSuite) TestCompleteCredentialsFromEnvKeypairCompatibleEn
 		"OS_PASSWORD":     "",
 		"NOVA_API_KEY":    "test-access-key",
 		"EC2_SECRET_KEYS": "test-secret-key",
-		"OS_TENANT_NAME":  "tenant-name",
+		"OS_TENANT_ID":    "tenant-name",
```
Comment (wallyworld) [inline identity/identity_test.go]: ditto - either _NAME or _ID needs to work
just renaming this doesn't test that

File: identity/identity.go (hunk 1 above)
Comment (ExternalReality) [reply identity/identity.go]: Err, sorry, forgot to remove that.

**Review by wallyworld — APPROVED** [review]

**Comment by ExternalReality** [comment]
$$Merge$$

**Comment by ExternalReality** [comment]
!!build!!

**Comment by ExternalReality** [comment]
$$Merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**